File: LeetCode/0001_Two_Sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookupTable = {}
        for i, n in enumerate(nums):
            diff = target - n
            if diff in lookupTable:
                return [lookupTable[diff], i]
            else:
                lookupTable[n] = i

File: LeetCode/0001_Two_Sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const lookupTable = {};
    for(let i = 0; i < nums.length; i++){
        const diff = target - nums[i];
        if(diff in lookupTable){
            return [lookupTable[diff], i];
        }
        lookupTable[nums[i]] = i;
    }
};
